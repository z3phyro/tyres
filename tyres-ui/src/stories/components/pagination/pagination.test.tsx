import { expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import Pagination from "./pagination";

test("Pagination renders", () => {
  const { getByRole } = render(() => <Pagination count={10} />);

  const pagination = getByRole("navigation");
  expect(pagination).toBeTruthy();
});

test("Pagination renders with custom count", () => {
  const { getAllByRole } = render(() => <Pagination count={3} />);

  const pagination = getAllByRole("listitem");
  expect(pagination.length).toBe(5);
});

test("Pagination snapshot test", () => {
  const { asFragment } = render(() => <Pagination count={10} />);

  expect(asFragment()).toMatchSnapshot();
});

