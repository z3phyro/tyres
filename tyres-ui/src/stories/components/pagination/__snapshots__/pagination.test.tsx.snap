// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination snapshot test 1`] = `"<nav id="pagination-cl-2" class="[&amp;>ul]:flex [&amp;>ul]:items-center [&amp;>ul]:justify-center [&amp;>ul]:gap-x-1 [&amp;>ul]:mt-2 "><ul><li><button type="button" disabled="" tabindex="-1" aria-disabled="true" data-disabled="" class="min-h-[38px] min-w-[38px] pyf-2 px-2.5 inline-flex justify-center items-center gap-x-2 text-sm rounded-lg border border-transparent text-gray-800 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 disabled:opacity-50 disabled:pointer-events-none dark:border-transparent dark:text-white dark:hover:bg-white/10 dark:focus:bg-white/10"><svg class="flex-shrink-0 size-3.5" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="m15 18-6-6 6-6"></path></svg><span aria-hidden="true" class="sr-only">Previous</span></button></li><li><button type="button" aria-current="page" data-current="" class="min-h-[38px] min-w-[38px] flex justify-center items-center border border-gray-200 text-gray-800 py-2 px-3 text-sm rounded-lg focus:outline-none focus:bg-gray-50 disabled:opacity-50 disabled:pointer-events-none dark:border-neutral-700 dark:text-white dark:focus:bg-white/10 border-gray-200 focus:bg-gray-50 disabled:opacity-50 disabled:pointer-events-none dark:border-neutral-700 dark:text-white dark:focus:bg-white/10">1</button></li><li><button type="button" class="min-h-[38px] min-w-[38px] flex justify-center items-center border border-gray-200 text-gray-800 py-2 px-3 text-sm rounded-lg focus:outline-none focus:bg-gray-50 disabled:opacity-50 disabled:pointer-events-none dark:border-neutral-700 dark:text-white dark:focus:bg-white/10 border-transparent hover:bg-gray-100 focus:bg-gray-100 disabled:opacity-50 disabled:pointer-events-none dark:border-transparent dark:text-white dark:hover:bg-white/10 dark:focus:bg-white/10">2</button></li><li><div class="min-h-[38px] min-w-[38px] flex justify-center items-center border border-transparent text-gray-800 hover:bg-gray-100 py-2 px-3 text-sm rounded-lg focus:outline-none disabled:opacity-50 disabled:pointer-events-none"> ...</div></li><li><button type="button" class="min-h-[38px] min-w-[38px] flex justify-center items-center border border-gray-200 text-gray-800 py-2 px-3 text-sm rounded-lg focus:outline-none focus:bg-gray-50 disabled:opacity-50 disabled:pointer-events-none dark:border-neutral-700 dark:text-white dark:focus:bg-white/10 border-transparent hover:bg-gray-100 focus:bg-gray-100 disabled:opacity-50 disabled:pointer-events-none dark:border-transparent dark:text-white dark:hover:bg-white/10 dark:focus:bg-white/10">10</button></li><li><button type="button" class="min-h-[38px] min-w-[38px] py-2 px-2.5 inline-flex justify-center items-center gap-x-2 text-sm rounded-lg border border-transparent text-gray-800 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 disabled:opacity-50 disabled:pointer-events-none dark:border-transparent dark:text-white dark:hover:bg-white/10 dark:focus:bg-white/10"><span aria-hidden="true" class="sr-only">Next</span><svg class="flex-shrink-0 size-3.5" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="m9 18 6-6-6-6"></path></svg></button></li></ul></nav>"`;
